File: Sources/ReusableOnboarding/OnboardingPage.swift
import Foundation

public struct OnboardingPage: Identifiable, Equatable, Sendable {
    public let id: String
    public let title: String
    public let subtitle: String
    public let systemImage: String
    public let accentLabel: String

    public init(
        id: String,
        title: String,
        subtitle: String,
        systemImage: String,
        accentLabel: String
    ) {
        self.id = id
        self.title = title
        self.subtitle = subtitle
        self.systemImage = systemImage
        self.accentLabel = accentLabel
    }
}

File: onboarding-android/src/main/kotlin/com/vadimkomis/onboarding/OnboardingPage.kt
package com.vadimkomis.onboarding

import android.net.Uri
import androidx.annotation.DrawableRes
import androidx.compose.runtime.Immutable
import androidx.compose.ui.graphics.vector.ImageVector

@Immutable
public data class OnboardingPage(
    public val id: String,
    public val title: String,
    public val subtitle: String,
    public val media: OnboardingPageMedia,
    public val accentLabel: String,
)

@Immutable
public sealed interface OnboardingPageMedia {
    public val contentDescription: String?

    public data class Icon(
        public val imageVector: ImageVector,
        public override val contentDescription: String? = null,
    ) : OnboardingPageMedia

    public data class Drawable(
        @param:DrawableRes public val resourceId: Int,
        public override val contentDescription: String? = null,
    ) : OnboardingPageMedia

    public data class Video(
        public val uri: Uri,
        public override val contentDescription: String? = null,
    ) : OnboardingPageMedia
}
